packages/amplify-graphql-transformer-core/src/transformation/validation.ts

```diff
@@ -160,7 +159,13 @@ export const validateAuthModes = (authConfig: AppSyncAuthConfiguration) => {
   for (let i = 0; i < authModes.length; i++) {
     const mode = authModes[i];
 
-    if (mode !== 'API_KEY' && mode !== 'AMAZON_COGNITO_USER_POOLS' && mode !== 'AWS_IAM' && mode !== 'OPENID_CONNECT' && mode !== 'AWS_LAMBDA') {
+    if (
+      mode !== 'API_KEY' &&
+      mode !== 'AMAZON_COGNITO_USER_POOLS' &&
+      mode !== 'AWS_IAM' &&
+      mode !== 'OPENID_CONNECT' &&
+      mode !== 'AWS_LAMBDA'
+    ) {
       throw new Error(`Invalid auth mode ${mode}`);
     }
   }
```

Comment:
nit: I know this was just a formatting change, but this could be simplified a bit to:

```js
const allowedAuthModes = ['API_KEY', 'AWS_IAM', ...];
const invalidMode = authModes.find(mode => !allowedAuthModes.includes(mode));
if (!!invalidMode) {
  throw new Error(`Invalid auth mode ${mode}`)
}
```

Reply:
If we're going to go this route, please hoist the array out of the function so that the same array can be reused. It might also make sense to use a Set instead of an array (not sure if there are enough elements to make it worth it though).